jetty-xml/src/main/java/org/eclipse/jetty/xml/ConfigurationProcessor.java

```diff
@@ -31,8 +33,14 @@
  */
 public interface ConfigurationProcessor
 {
-    public void init(URL url, XmlParser.Node root, XmlConfiguration configuration);
-    
-    public Object configure( Object obj) throws Exception;
-    public Object configure() throws Exception;
+    /**
+     * @deprecated use {@link #init(Resource, XmlParser.Node, XmlConfiguration)} instead
+     */
+    @Deprecated
+    void init(URL url, XmlParser.Node root, XmlConfiguration configuration);
+
+    void init(Resource resource, XmlParser.Node root, XmlConfiguration configuration);
```

Comment:
I don't think we need to modify the `init()` method, see above.
Even if we do, it's better to implement the deprecated one here as a `default` method to call the new one.

Reply:
This is the source of bugs, especially with escaping on Windows.
If this had been URI then we have no problem.

URL should be deprecated, Resource should be the only mechanism.

Reply:
URL is deprecated and set as default for Resource version.
But I think this might be backwards?

Reply:
Resolved.